sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param data The manifest that needs to be uploaded.
+     * @return The operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public UploadManifestResult uploadManifest(BinaryData data) {
+        return this.asyncClient.uploadManifest(data).block();
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param data The manifest that needs to be uploaded.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return The rest response containing the operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<UploadManifestResult> uploadManifestWithResponse(BinaryData data, Context context) {
+        return this.asyncClient.uploadManifestWithResponse(data.toFluxByteBuffer(), context).block();
+    }
+
+    /**
+     * Uploads a blob to the repository.
+     * The client currently uploads the entire blob\layer as a single unit.
+     * <p>
+     * Also, the blob is read into memory and then an upload operation is performed as a single operation.
+     * We currently do not support breaking the layer into multiple chunks and uploading them one at a time
+     *
+     * @param data The blob\image content that needs to be uploaded.
+     * @return The operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public UploadBlobResult uploadBlob(BinaryData data) {
```

Comment:
This is something we discussed in FC and blob seemed the right term here to give us more flexibility for the following scenario 

Today blob is same as image. We do not allow uploading part of the image. However in future we can support chunking since the rest API does. So blob is for future proofing.